Zend/zend_compile.c

```diff
@@ -2867,8 +2891,17 @@ static void zend_compile_list_assign(
 
 		zend_verify_list_assign_target(var_ast, old_style);
 
-		zend_emit_op(&fetch_result, ZEND_FETCH_LIST, expr_node, &dim_node);
-		zend_emit_assign_znode(var_ast, &fetch_result);
+		opline = zend_emit_op(&fetch_result, ZEND_FETCH_LIST, expr_node, &dim_node);
+
+		if (elem_ast->attr || (var_ast->kind == ZEND_AST_ARRAY && zend_set_list_assign_reference(var_ast))) {
+			opline->extended_value = ZEND_LIST_MAKE_WRITABLE;
+		}
+
+		if (elem_ast->attr) {
+			zend_emit_assign_ref_znode(var_ast, &fetch_result);
+		} else {
+			zend_emit_assign_znode(var_ast, &fetch_result);
+		}
```

Comment:
The `$a =& 5` is prohibited as a parse error, whereas shockingly `$a =& $x+$y` is executed, albeit I concur it probably shouldn't be.

```
<?php
$a = 10;
$b = 20;

$c =& $a + $b;

outputs:
int(10)
int(20)
int(10)
```

However, wrapping it like `$c =& ($a + $b)` is also a parse error.